src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,35 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder();
+        int employSalary;
+        LocalDate localDateTo = LocalDate.parse(dateTo, FORMATTER);
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, FORMATTER);
+        String[] parsedEmployData;
+        LocalDate localEmployDate;
+        report.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        for (String employName : names) {
+            employSalary = 0;
+            for (String employData : data) {
+                parsedEmployData = employData.split(" ");
+                if (employName.equals(parsedEmployData[1])) {
```

Comment:
Please, move 0, 1, 2, 3 indexes to constants. It can be:
DATE_INDEX, NAME_INDEX, HOURS_INDEX, SALARY_PER_HOUR_INDEX